static/app/views/dashboards/metrics/table.tsx

```diff
@@ -0,0 +1,212 @@
+  data: {
+    headers: {name: string; type: string}[];
+    rows: any[];
+  };
+  isLoading: boolean;
+  borderless?: boolean;
+}
+
+export function MetricTable({isLoading, data, borderless}: MetricTableProps) {
+  function renderRow(row: any, index: number) {
+    return data.headers.map((column, columnIndex) => {
+      const key = `${index}-${columnIndex}:${column}`;
+      const value = row[column.name];
+      if (!value) {
+        return (
+          <TableCell type={column.type} key={key} noValue>
+            {column.type === 'field' ? 'n/a' : '(none)'}
+          </TableCell>
+        );
+      }
+      return (
+        <TableCell type={column.type} key={key}>
+          {value}
+        </TableCell>
+      );
+    });
+  }
+
+  return (
+    <StyledPanelTable
+      borderless={borderless}
+      headers={data.headers.map((column, index) => {
+        const header = formatMRIField(column.name);
+        return (
+          <HeaderCell key={index} type={column.type}>
+            <Tooltip title={header}>{header}</Tooltip>
+          </HeaderCell>
+        );
+      })}
+      isLoading={isLoading}
+      emptyMessage={t('No results')}
+    >
+      {data.rows.map(renderRow)}
+    </StyledPanelTable>
+  );
+}
+
+const groupBysMatch = (a: Record<string, any>, b: Record<string, any>) => {
+  return JSON.stringify(a) === JSON.stringify(b);
+};
+
+const getEmptyGroup = (tags: string[]) =>
+  tags.reduce((acc, tag) => {
+    acc[tag] = '';
+    return acc;
+  }, {});
+
+function getGroupByCombos(
+  queries: MetricsQueryApiRequestQuery[],
+  results: MetricsQueryApiResponse['data']
+): Record<string, string>[] {
+  const groupBys = Array.from(new Set(queries.flatMap(query => query.groupBy ?? [])));
+  const emptyBy = getEmptyGroup(groupBys);
+
+  const allCombos = results.flatMap(group => {
+    return group.map(entry => ({...emptyBy, ...entry.by}));
+  });
+
+  const uniqueCombos = allCombos.filter(
+    (combo, index, self) => index === self.findIndex(other => groupBysMatch(other, combo))
+  );
+
+  return uniqueCombos;
+}
+type Row = Record<string, string | undefined>;
+
+interface TableData {
+  headers: {name: string; type: string}[];
+  rows: Row[];
+}
+
+export function getTableData(
+  data: MetricsQueryApiResponse,
+  queries: MetricsQueryApiQueryParams[]
+): TableData {
+  const filteredQueries = queries.filter(
+    query => !isMetricFormula(query)
+  ) as MetricsQueryApiRequestQuery[];
+
+  const fields = filteredQueries.map(query => MRIToField(query.mri, query.op));
+  const tags = [...new Set(filteredQueries.flatMap(query => query.groupBy ?? []))];
+
+  const normalizedResults = filteredQueries.map((query, index) => {
+    const queryResults = data.data[index];
+    const normalizedGroupResults = queryResults.map(group => {
+      return {
+        by: {...getEmptyGroup(tags), ...group.by},
+        totals: formatMetricsUsingUnitAndOp(
+          group.totals,
+          parseMRI(query.mri)?.unit!,
```

Comment:
It would be better to also read the unit from the query response meta here.
Otherwise it will break once the BE PR that normalizes values is merged.